### ISSUE TITLE
No way to load all projects when a solution file is present

### ISSUE THREAD

**Issue by ahanusa** [dotnet/vscode-csharp#6565]
Type: <b>Bug</b>

## Issue Description ##

Go to definition via right click or F12 does not work when upgrading v 1.26 to 2.6.24

Video clip: https://www.dropbox.com/scl/fi/b3h4f7whrwin7t6y2be2x/Screen-Recording-2023-10-18-at-11.04.52-AM.mov?rlkey=7ie4mp8tqdud26eghdem9riou&dl=0

## Steps to Reproduce ##

1. With version 1.26 installed, open a folder containing many projects. Note: there is no solution file.
1. Open a class file that references another class/interface located in another directory and/or project
1. Click on the referenced class/interface and hit F12 or right click/Go to Definition to navigate to the file.

## Expected Behavior ##

You should be taken to the definition/file

## Actual Behavior ##

1. Now upgrade to v2.6.24, reload vs code, and repeat the repro steps

You will not be taken to the file and a message is displayed:
"No definition found for XXX".

### C# log ###
<details>Post the output from Output-->C# here</details>

### C# LSP Trace Logs ###
<details>Post the output from Output-->C# LSP Trace Logs here.  Requires `dotnet.server.trace` to be set to `Trace`</details>

## Environment information ##

**VSCode version**: 1.83.1
**C# Extension**: 2.6.24
**Using OmniSharp**: false

<details><summary>Dotnet Information</summary>
.NET SDK:
 Version:   7.0.307
 Commit:    7b77e35a2b

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  14.0
 OS Platform: Darwin
 RID:         osx-x64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.307/

Host:
  Version:      7.0.10
  Architecture: x64
  Commit:       a6dbb800a4

.NET SDKs installed:
  3.1.418 [/usr/local/share/dotnet/sdk]
  3.1.419 [/usr/local/share/dotnet/sdk]
  3.1.423 [/usr/local/share/dotnet/sdk]
  3.1.425 [/usr/local/share/dotnet/sdk]
  3.1.426 [/usr/local/share/dotnet/sdk]
  6.0.201 [/usr/local/share/dotnet/sdk]
  6.0.202 [/usr/local/share/dotnet/sdk]
  6.0.203 [/usr/local/share/dotnet/sdk]
  6.0.401 [/usr/local/share/dotnet/sdk]
  6.0.403 [/usr/local/share/dotnet/sdk]
  6.0.405 [/usr/local/share/dotnet/sdk]
  6.0.413 [/usr/local/share/dotnet/sdk]
  7.0.100 [/usr/local/share/dotnet/sdk]
  7.0.102 [/usr/local/share/dotnet/sdk]
  7.0.307 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.24 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.25 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.29 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.31 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.3 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.9 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.13 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.24 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.25 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.29 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.31 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.3 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.9 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|birds-of-paradise|Programming-Engineer|0.1.2|programming-engineer.birds-of-paradise-0.1.2|
|blazorwasm-companion|ms-dotnettools|1.1.4|ms-dotnettools.blazorwasm-companion-1.1.4|
|csharp|ms-dotnettools|2.6.24|ms-dotnettools.csharp-2.6.24-darwin-x64|
|csharpextensions|jchannon|1.3.1|jchannon.csharpextensions-1.3.1|
|EditorConfig|EditorConfig|0.16.4|editorconfig.editorconfig-0.16.4|
|gitblame|waderyan|10.5.1|waderyan.gitblame-10.5.1|
|hexeditor|ms-vscode|1.9.12|ms-vscode.hexeditor-1.9.12|
|json|Meezilla|0.1.2|meezilla.json-0.1.2|
|ng-template|Angular|16.1.8|angular.ng-template-16.1.8|
|nugetpackagemanagergui|aliasadidev|2.0.6|aliasadidev.nugetpackagemanagergui-2.0.6|
|path-intellisense|christian-kohler|2.8.5|christian-kohler.path-intellisense-2.8.5|
|powershell|ms-vscode|2023.8.0|ms-vscode.powershell-2023.8.0|
|remote-containers|ms-vscode-remote|0.315.1|ms-vscode-remote.remote-containers-0.315.1|
|Theme-BirdsofParadise|gerane|0.0.5|gerane.theme-birdsofparadise-0.0.5|
|theme-github|thomaspink|1.0.1|thomaspink.theme-github-1.0.1|
|Theme-peacocks-in-space|gerane|0.0.5|gerane.theme-peacocks-in-space-0.0.5|
|trailing-spaces|shardulm94|0.4.1|shardulm94.trailing-spaces-0.4.1|
|vim|vscodevim|1.26.1|vscodevim.vim-1.26.1|
|vscode-commons|redhat|0.0.6|redhat.vscode-commons-0.0.6|
|vscode-docker|ms-azuretools|1.27.0|ms-azuretools.vscode-docker-1.27.0|
|vscode-dotnet-runtime|ms-dotnettools|1.8.1|ms-dotnettools.vscode-dotnet-runtime-1.8.1|
|vscode-guid|heaths|1.7.0|heaths.vscode-guid-1.7.0|
|vscode-nuget-gallery|patcx|0.0.24|patcx.vscode-nuget-gallery-0.0.24|
|vscode-resx-editor|DominicVonk|0.0.5|dominicvonk.vscode-resx-editor-0.0.5|
|vscode-theme-onelight|akamud|2.3.0|akamud.vscode-theme-onelight-2.3.0|
|vscode-yaml|redhat|1.14.0|redhat.vscode-yaml-1.14.0|;

</details>

Extension version: 2.6.24
VS Code version: Code 1.83.1 (f1b07bd25dfad64b0167beb15359ae573aecd2cc, 2023-10-10T23:46:55.789Z)
OS version: Darwin x64 23.0.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|6, 4, 4|
|Memory (System)|16.00GB (0.28GB free)|
|Process Argv|--crash-reporter-id ab71c575-1d1a-4ad3-88ac-549c056db4b2 --crash-reporter-id ab71c575-1d1a-4ad3-88ac-549c056db4b2|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30856252
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
2e4cg342:30602488
f6dab269:30613381
showlangstatbar:30737416
962ge761:30841074
03d35959:30757346
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxptcf:30805731
pythonnoceb:30805159
copilotsettingt:30859503
dsvsc013:30795093
dsvsc014:30804076
diffeditorv2:30821572
pythonmpsinfo:30842941
dsvsc015:30845448
pyreplss2:30858576
pythoncet0:30859751
pythontbext0:30864172

```

</details>

**Comment by dibarbet** [comment]
@ahanusa can you post the C# output window contents here?  This might be a known issue

**Comment by ahanusa** [comment]
Here is the output as-is:

`
Using dotnet configured on PATH
Dotnet path: /usr/local/share/dotnet/dotnet
Activating C# standalone...
waiting for named pipe information from server...
[stdout] {"pipeName":"/var/folders/92/bxy3961s0c39s5d2r5mfs8cw0000gp/T/eae4447e-c631-456c-8e5a-16b847e9c823.sock"}
received named pipe information from server
attempting to connect client to server...
client has connected to server
[Info  - 1:02:36 PM] [Program] Language server initialized
`

Where would i set `dotnet.server.trace`?

**Comment by dibarbet** [comment]
@ahanusa in vscode settings - 
<img width="957" alt="image" src="https://github.com/dotnet/vscode-csharp/assets/5749229/33c656c4-fa09-4c23-9863-ef0c130f8003">

@jasonmalinowski looks like everything is in misc and theres no project loading lines.  Maybe something going wrong on the client side with finding all projects to load?

**Comment by ahanusa** [comment]
@dibarbet I updated my last message.  I figured out how to trace and re-ran before your reply.  Sorry for the confusion.

**Comment by jasonmalinowski** [comment]
@ahanusa What types of projects do you have in this workspace? Are the .csproj files actually present in the workspace?

**Comment by jasonmalinowski** [comment]
@ahanusa Also, if you go through your options prefixed with either 'dotnet' or 'csharp', what is set?

**Comment by ahanusa** [comment]
> @ahanusa What types of projects do you have in this workspace? Are the .csproj files actually present in the workspace?

I've never setup a workspace.  The way I've always worked is by opening a directory containing all of the projects directly

**Comment by ahanusa** [comment]
> @ahanusa Also, if you go through your options prefixed with either 'dotnet' or 'csharp', what is set?

What "options" are you referring to?

**Comment by jasonmalinowski** [comment]
Ah yes, remove the "dotnet.defaultSolution" thing -- if that's set to disable that's going to cause us to disable loading.

**Comment by ahanusa** [comment]
Ok, so I received a message when reopening that I thought was in-error, but clicking no re-added that option to my workspace settings:

<img width="573" alt="Screenshot 2023-10-18 at 1 46 45 PM" src="https://github.com/dotnet/vscode-csharp/assets/1224177/a9e493df-d1dc-45a5-b0cf-d719fa52f311">

Turns out there are a couple solution files in some of the folders.  However, they belong to another team and have no use to me.

**Comment by ahanusa** [comment]
@jasonmalinowski any other thoughts?

**Comment by jasonmalinowski** [comment]
@ahanusa At this point this is a limitation of the new extension; we don't have a way to say "I don't want to load a solution, just load all the projects". The workaround for now would be to check in a new solution file with whatever projects you do want. You can use [dotnet sln](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-sln) to make the solution file, there's some examples of how to pass in globbing if you just wanted to add all .csproj files to your solution in one single go.

Can you share what your Omnisharp settings were? Did you have load on demand enabled? Or if not, where you manually picking projects before?

**Comment by ahanusa** [comment]
> @ahanusa At this point this is a limitation of the new extension; we don't have a way to say "I don't want to load a solution, just load all the projects". The workaround for now would be to check in a new solution file with whatever projects you do want. You can use [dotnet sln](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-sln) to make the solution file, there's some examples of how to pass in globbing if you just wanted to add all .csproj files to your solution in one single go.
> 
> Can you share what your Omnisharp settings were? Did you have load on demand enabled? Or if not, where you manually picking projects before?

Ok, that seemed to work.  Thanks for the help and timely correspondence!

**Comment by jasonmalinowski** [comment]
@ahanusa Happy to hear you're unblocked! We'll still leave this open since there is an implicit feature request to make this scenario a bit easier.

**Comment by ahanusa** [comment]
@jasonmalinowski I just noticed after getting everything resolved that the `run` and `debug` links that used to hover above unit test methods are no longer present?

**Comment by ahanusa** [comment]
@jasonmalinowski After upgrading to latest C# extension the `run` and `debug` links no longer appear above unit test methods?

**Comment by dibarbet** [comment]
Closing - feature for loading all projects w/out sln is tracked by https://github.com/dotnet/vscode-csharp/issues/5722 and https://github.com/dotnet/vscode-csharp/issues/6008

> After upgrading to latest C# extension the run and debug links no longer appear above unit test methods?

Should have been resolved in latest versions, see https://github.com/dotnet/vscode-csharp/issues/5719.  If you're having issues, please file a new issue.
